pkg/routing/config_map_updater.go

```diff
@@ -21,51 +17,35 @@ import (
 //	called ConfigMapRoutingTableName. On either of those events, decodes
 //	that ConfigMap into a routing table and stores the new table into table
 //	using table.Replace(newTable)
+//	- Execute the callback function, if exist
 //	- Returns an appropriate non-nil error if ctx.Done() receives
 func StartConfigMapRoutingTableUpdater(
 	ctx context.Context,
 	lggr logr.Logger,
-	updateEvery time.Duration,
-	getterWatcher k8s.ConfigMapGetterWatcher,
+	cmInformer *k8s.InformerConfigMap,
+	ns string,
 	table *Table,
 	q queue.Counter,
+	cbFunc func() error,
 ) error {
 	lggr = lggr.WithName("pkg.routing.StartConfigMapRoutingTableUpdater")
-	watchIface, err := getterWatcher.Watch(ctx, metav1.ListOptions{})
-	if err != nil {
-		return err
-	}
-	defer watchIface.Stop()
 
-	ticker := time.NewTicker(updateEvery)
-	defer ticker.Stop()
+	informerMsg := make(chan error)
+	go func() {
+		defer ctx.Done()
+		err := cmInformer.Start(ctx)
+		if err != nil {
+			informerMsg <- err
+		}
+	}()
```

Comment:
do you think it would be worth using an [errgroup](https://pkg.go.dev/golang.org/x/sync/errgroup) for this

Reply:
I also think errgroup would be more appropriate